### ISSUE TITLE
`--target rollup` option effect on linked libraries (`--link-library` option)

### ISSUE THREAD

**Issue by greenhat** [0xMiden/compiler#355]
I was quite surprised that `--target rollup` only adds `-link-library` for `std` and `base` only if no other `--link-library` option is specified. As soon as I add custom library linking via `--link-library` the `std` and `base` disappear and I have to add them manually. Is it a bug or intended behavior?

**Comment by greenhat** [comment]
I believe that it happens at https://github.com/0xPolygonMiden/compiler/blob/17aa362d8a9e76cc8c2d30a01f8fb0cd6fb9fa5a/midenc-debug/src/cli.rs#L72

**Comment by bitwalker** [comment]
Yeah, setting `link-library` options when specifying certain `--target` options is a bit of a hack, I'm not sure we can make the combination of flags additive. We probably need to explicitly check if both options are provided, and if so, ensure that all of the implied link libraries for the given `--target` are still present (adding them if not).
